crnk-core/src/main/java/io/crnk/core/queryspec/QuerySpec.java

```diff
@@ -261,16 +266,20 @@ public void setIncludedRelations(List<IncludeRelationSpec> includedRelations) {
     }
 
     public Collection<QuerySpec> getNestedSpecs() {
-        return Collections.unmodifiableCollection(relatedSpecs.values());
+    	// Using a set to remove duplicate querySpec between typeRelatedSpecs and classRelatedSpecs
+    	Set<QuerySpec> allRelatedSpecs = new HashSet(typeRelatedSpecs.values());
+    	allRelatedSpecs.addAll(classRelatedSpecs.values());
+        return Collections.unmodifiableCollection(allRelatedSpecs);
     }
 
     public void setNestedSpecs(Collection<QuerySpec> specs) {
-        this.relatedSpecs.clear();
+    	this.typeRelatedSpecs.clear();;
+        this.classRelatedSpecs.clear();
         for (QuerySpec spec : specs) {
             if (spec.getResourceClass() != null) {
```

Comment:
I suspect here the getResourceType() should have priority?

Reply:
Yes you're right. I've just updated the pull request to reflect the changes